Modules/Quest/QuestieQuest.lua

```diff
@@ -1279,65 +1287,65 @@ function _QuestieQuest:GetQuestIcon(quest)
     return icon
 end
 
----@type Ticker?
-local timer
-function QuestieQuest:CalculateAndDrawAvailableQuestsIterative(callback)
-    Questie:Debug(Questie.DEBUG_INFO, "[QuestieQuest:CalculateAndDrawAvailableQuestsIterative] PlayerLevel =", QuestiePlayer.GetPlayerLevel())
-
-    -- Localize the variable for speeeeed
-    local debugEnabled = Questie.db.global.debugEnabled
+--? Creates a localized space where the local variables and functions are stored
+do
```

Comment:
Why is this needed? 🤔 

Top level logic is scary

Reply:
I don't really understand?
The do block is used to not enable anything outside of it to be able to modify the timer variable which was more global before.
The only logic that escapes the do-block scope is the actual function that gets created into QuestieQuest but the local function and variable are only available within scope by design.